src/R/Editor/Impl/Tree/TextChangeAnalyzer.cs

```diff
@@ -64,6 +68,34 @@ internal static class TextChangeAnalyzer {
             return TextChangeType.Structure;
         }
 
+        private static bool IsLineBreakSensitive(TextChangeContext context, IAstNode node) {
+            if (node is If) {
+                return true;
+            }
+
+            var position = context.PendingChanges.Start;
+            var candidate = node.Root.GetNodeOfTypeFromPosition<If>(position);
+            if (candidate != null) {
+                return true;
+            }
+
+            // Check if line break is added to removed on a line with closing } of 'if'
+            // so we can full-parse if 'else' position changes relatively to the if
+            var snapshot = context.EditorTree.EditorBuffer.CurrentSnapshot;
+            var text = snapshot.GetLineFromPosition(context.PendingChanges.Start).GetText();
+
+            // We need to find if any position in the line belong to `if` scope
+            // if there is 'else' the same line
+            return FindKeyword("if", text) || FindKeyword("else", text);
```

Comment:
`RTokenizer` can be reused (e.g. `FindAnyKeyword(text, "if", "else")`). Does it matter in terms of performance?